init.lua

```diff
@@ -103,33 +102,43 @@ end
 -- Sprint settings and function
 local enable_sprint = minetest.setting_getbool("sprint") ~= false
 local enable_sprint_particles = minetest.setting_getbool("sprint_particles") ~= false
+local player_monoids_mod = minetest.get_modpath("player_monoids")
 local armor_mod = minetest.get_modpath("3d_armor")
 
 function set_sprinting(name, sprinting)
 	local player = minetest.get_player_by_name(name)
 	local def = {}
-	-- Get player physics from 3d_armor mod
-	if armor_mod and armor and armor.def then
-		def.speed = armor.def[name].speed
-		def.jump = armor.def[name].jump
-		def.gravity = armor.def[name].gravity
-	end
+	if player_monoids_mod and player_monoids then
+		if sprinting == true then
+			player_monoids.speed:add_change(player, 1 + SPRINT_SPEED, "stamina:physics")
+			player_monoids.jump:add_change(player, 1 + SPRINT_JUMP, "stamina:physics")
```

Comment:
How about gravity?

Reply:
stamina doesn't affect gravity, does it?

Reply:
Stamina not directly, but stamina supports armor (see armor_mod checks) which might use different gravity values.

Reply:
Well we're not overriding any gravity values this way. 3d_armor supports player_monoids for speed and jump too so we don't need to manually insert them in order to remain compatible when using this API.